onnx/reference/custom_element_types.py

```diff
@@ -5,10 +5,72 @@
 
 import numpy as np
 
+try:
+    import ml_dtypes
+except ImportError:
+    ml_dtypes = None  # type: ignore[assignment]
+
 bfloat16 = np.dtype((np.uint16, {"bfloat16": (np.uint16, 0)}))
 float8e4m3fn = np.dtype((np.uint8, {"e4m3fn": (np.uint8, 0)}))
 float8e4m3fnuz = np.dtype((np.uint8, {"e4m3fnuz": (np.uint8, 0)}))
 float8e5m2 = np.dtype((np.uint8, {"e5m2": (np.uint8, 0)}))
 float8e5m2fnuz = np.dtype((np.uint8, {"e5m2fnuz": (np.uint8, 0)}))
 uint4 = np.dtype((np.uint8, {"uint4": (np.uint8, 0)}))
 int4 = np.dtype((np.int8, {"int4": (np.int8, 0)}))
+
+
+def convert_from_ml_dtypes(tensor: np.ndarray) -> np.ndarray:
```

Comment:
```suggestion
def convert_from_ml_dtypes(array: np.ndarray) -> np.ndarray:
```

NBD but more conventional for np arrays